sdk/attestation/azure-security-attestation/azure/security/attestation/_common.py

```diff
@@ -5,27 +5,108 @@
 # Code generated by Microsoft (R) AutoRest Code Generator.
 # Changes may cause incorrect behavior and will be lost if the code is regenerated.
 # --------------------------------------------------------------------------
-from typing import TYPE_CHECKING
+from typing import Dict, TYPE_CHECKING, Tuple
+import base64
+
+from cryptography.hazmat.backends import default_backend
+from cryptography.hazmat.primitives import serialization
+from cryptography.hazmat.primitives.asymmetric.ec import EllipticCurvePrivateKey
+from cryptography.hazmat.primitives.asymmetric.rsa import RSAPrivateKey
+from cryptography.x509.base import load_pem_x509_certificate
+from cryptography.x509 import Certificate
+from cryptography.hazmat.primitives.serialization import Encoding, PublicFormat
 
 if TYPE_CHECKING:
     # pylint: disable=unused-import,ungrouped-imports
     from typing import Any
 
 
-import base64
+def base64url_encode(unencoded):
+    # type: (bytes) -> str
+    base64val = base64.urlsafe_b64encode(unencoded)
+    strip_trailing = base64val.split(b"=")[0]  # pick the string before the trailing =
+    return strip_trailing.decode("utf-8")
+
+
+def base64url_decode(encoded):
+    # type: (str) -> bytes
+    padding_added = encoded + "=" * ((len(encoded) * -1) % 4)
+    return base64.urlsafe_b64decode(padding_added.encode("utf-8"))
+
+
+def pem_from_base64(base64_value, header_type):
+    # type: (str, str) -> str
+    pem = "-----BEGIN " + header_type + "-----\n"
+    while base64_value != "":
+        pem += base64_value[:64] + "\n"
+        base64_value = base64_value[64:]
+    pem += "-----END " + header_type + "-----\n"
+    return pem
+
+
+def validate_signing_keys(signing_key_pem, certificate_pem):
+    # type: (str, str) -> Tuple[EllipticCurvePrivateKey, RSAPrivateKey, Certificate]
+    """Validates the attestation signing key and certificates specified.
+
+    :param str signing_key_pem: PEM encoded EC or RSA signing key.
+    :param str certificate_pem: PEM encoded X.509 certificate.
+    :return: Returns the decoded signing key and certificate
+    :rtype: RSAPrivateKey or ElilipticCurvePrivateKey, Certificate
 
-class Base64Url:
-    """Equivalent to base64.urlsafe_b64encode, but strips padding from the encoded and decoded strings.
+    The validate_signing_keys method decodes the signing key and certificate
+    and verifies that the public key associated with the certificate and the
+    signing key are the same key.
+
+    :staticmethod:
```

Comment:
I'm assuming the end of this docstring was accidental, unless I'm missing context
```suggestion
    signing key are the same key.
```